doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -668,69 +652,83 @@ Induction
       Use the :cmd:`Add` and :cmd:`Remove` commands to update this set manually.
 
-.. tacn:: inversion @ident
-   :name: inversion
+:tacn:`inversion`
+~~~~~~~~~~~~~~~~~
+
+.. tacn:: inversion {| @ident | @natural } using @one_term {? in {+ @ident } }
+          inversion {| @ident | @natural } {? as @or_and_intropattern } {? in {+ @ident } }
+   :name: inversion; _
+
+   .. comment: the other inversion* tactics don't support the using clause,
+      but they should be able to, if desired.  It wouldn't make sense for
+      inversion_sigma.
+      See https://github.com/coq/coq/pull/14179#discussion_r642193096
+
+   .. todo PR: following is clear as mud
 
    Let the type of :n:`@ident` in the local context be :g:`(I t)`, where :g:`I`
-   is a (co)inductive predicate. Then, ``inversion`` applied to :n:`@ident`
+   is a (co-)inductive predicate. Then, ``inversion`` applied to :n:`@ident`
    derives for each possible constructor :g:`c i` of :g:`(I t)`, all the
    necessary conditions that should hold for the instance :g:`(I t)` to be
    proved by :g:`c i`.
 
+   :n:`@ident`
+     The name of the hypothesis to apply.
+
+   :n:`@natural`
+     Equivalent to :n:`intros until @natural; inversion @ident`
+     where :n:`@ident` is the identifier for the last introduced hypothesis.
+
+   :n:`{? in {+ @ident } }`
+     When :n:`{+ @ident}` are identifiers in the local context, this causes
+     generalizing on :n:`{+ @ident}` as the initial step of `inversion`.
+     See :tacn:`generalize`.
+
+   .. todo PR: "generalizing"?  As in :n:`generalize @ident__1; ... generalize @ident__n` ?
+
+   :n:`as @or_and_intropattern`
+      Provides names for the variables introduced in each new subgoal.  The
+      :token:`or_and_intropattern` must have one :n:`{* @intropattern }`
+      for each constructor of the type of :n:`(I t)`, given in the order
+      in which the constructors are defined.
+      If there are not enough names, Coq picks fresh names.
+
+     If an equation splits into several
+     equations (because ``inversion`` applies ``injection`` on the equalities it
+     generates), the corresponding :n:`@intropattern` should be in the form of
+     :n:`@or_and_intropattern` (or the equivalent :n:`{*, ( @simple_intropattern ) }`),
+     with the number of entries equal to the number
+     of subequalities obtained from splitting the original equation.
+
+     Here is an example: the ``inversion … as`` variant of
+     ``inversion`` generally behaves in a slightly more expected way than
+     ``inversion`` (no artificial duplication of some hypotheses referring to
+     other hypotheses). To take advantage of these improvements, it is enough to use
+     ``inversion … as []``, letting Coq choose fresh names.
+
 .. note::
    If :n:`@ident` does not denote a hypothesis in the local context but
    refers to a hypothesis quantified in the goal, then the latter is
    first introduced in the local context using :n:`intros until @ident`.
```

Comment:
I have the same doubts about the validity of this use case, and generally the same remarks regarding this as for `discriminate` and `injection`.

Reply:
Would you be willing to try it and see if it's accurate or not?